### DISCUSSION TITLE
Display multiple plots in one cell

### DISCUSSION THREAD

**Discussion by simonmandlik** [fonsp/Pluto.jl#2197] [Questions about using Pluto] [answered]
Hi, is there any way to display multiple plots in one cell? Say I have a clustering procedure in which, depending on the data, I can obtain 2-20 clusters and want to visualize all of them separately. In jupyter notebook, I would use something like this

```julia
for cluster in clusters
    @info cluster
    plot(cluster) |> display
end;
```

And it would give me the correct amount of plots each time. In jupyter, I assume it is best to create a separate cell for each cluster, which is really inconvenient because (i) there is a variable number of clusters and (ii) the amount of cells would get quickly out of hand. What is the recommended way to deal with this? Thanks!

**Comment by simonmandlik** [comment]
Using subplotting is not an option here, because I want to also print out some statistics with PrettyTables.jl to stdout

**Comment by rikhuijzer** [comment, marked as answer]
Have you tried returning a vector of plots?
```julia
[plot(cluster) for cluster in clusters]
```
(so don't end this with a semicolon `;`)

**Reply by simonmandlik** [reply]
Thanks, that works, but is kind of an ugly workaround, isn't it?

<img width="709" alt="image" src="https://user-images.githubusercontent.com/16707112/176899098-e542fe12-24b2-4a9d-a211-f94c072239ce.png">

Jupyter in comparison

<img width="561" alt="image" src="https://user-images.githubusercontent.com/16707112/176900515-d318487d-9aab-4b7f-aaf9-b3d41a58df7e.png">

**Reply by rikhuijzer** [reply]
> Thanks, that works, but is kind of an ugly workaround, isn't it?

I wouldn't say so. Try printing thousands of plots. Jupyter will crash (?) or truncate on the lines since the only thing it has is lines. Pluto, instead, has seen the array and will truncate on full plots. So, Pluto will show the last and first plot probably. Also, if you print to text, then you loose the original plot whereas you can print an array of plots in Pluto quite easily and then re-use that array to read the plots and copy+modify them, for example.

**Reply by simonmandlik** [reply]
You are right, but what is the point of plotting thousands of plots? I am using interactive notebooks mainly for presenting or interactive manual data analysis. In such cases, the optimal amount of plots to look at before loosing attention / attention of audience is 2-20 as I have mentioned in the original question.

**Reply by rikhuijzer** [reply]
For what it's worth. This is Prof. Alan Edelman (MIT) showing images inside a vector: https://youtu.be/3zTO3LEY-cM?t=2899

**Reply by simonmandlik** [reply]
No doubts there are great uses of this. But the images in the lecture server a completely different purpose than what I'm trying to describe here

**Reply by icweaver** [reply]
I guess each plot could be placed in an info box, but I have to admit that the way it is currently laid out does make navigation a bit touchy:

```julia
let
    for i ∈ 1:10
        @info "Plot $i" plot(rand(10))
    end
end
```

https://user-images.githubusercontent.com/25312320/176925332-9a34f514-e2e3-41eb-aa3c-b09f1f727c80.mp4

**Reply by holomorphism** [reply]
For the sake of appearance, I prefer to use HypertextLiteral.

```julia
using HypertextLiteral

# example (1)
let
	clusters = [rand(10) for _ in 1:2]
	
	plots = []
	for (i,cluster) in enumerate(clusters)
		info = htl"Plot: $i"
		p = plot(cluster; size=(500,150))
		push!(plots, htl"<div>$info</div><div>$p</div>")
	end
	
	htl"<h5>(1)</h5>$plots"
end

# example (2)
let
	clusters = [rand(10) for _ in 1:10]
	
	plots = []
	for (i,cluster) in enumerate(clusters)
		#info = htl"Plot: $i"
		p = plot(cluster; size=(300,200))
		push!(plots, p)
	end
	
	htl"<h5>(2)</h5><div style='white-space: nowrap;'>$plots</div>"
end
```
![image](https://user-images.githubusercontent.com/3076585/177001849-c666f38d-c296-46c3-8f58-36f27531feff.png)
